src/guards.ts

```diff
@@ -107,10 +148,20 @@ export function isArray<T>(
  */
 export function isObject<T extends unknown>(
   data: T | object
-): data is T extends Record<string, unknown> ? T : never {
-  return !!data && !Array.isArray(data) && typeof data === 'object';
+): data is DefinitelyObject<T> {
+  return (
+    !!data &&
+    !Array.isArray(data) &&
+    !isPromise(data) &&
+    !isDate(data) &&
+    !isError(data) &&
```

Comment:
I changed the implementation here because I don't think that we'd want `isObject` to return `true` for a promise, a date or an error, but the `typeof` check does that at runtime. They are now also excluded on type level [here](https://github.com/remeda/remeda/pull/145/files#diff-d340d57cd15a4762ee2c5cdf7d5f3650908cd5c9ff483bd12e8130b22b99bac4R136).

This is debatable of course, but you'd mostly want to check for a "plain object", and you can check all other things with the other guards. If you're against this, we could keep `isObject` as it was, and maybe add another `isPlainObject`(like lodash also has it) that does this check. Please let me know what you think.

Reply:
I think returning`true` for promises, dates, and errors is ok, and most people would expect that.  
Adding `isPlainObject` is ok.

Reply:
okay, I will change the implementation, add the additional `isPlainObject` guard and update the docs accordingly :)

Reply:
great, thanks :)

Reply:
done. let me know what you think

Reply:
Looks good, but I am not sure if we should add more checks? For example, add `!isRegex(data)`? Does it return false for custom classes?

Reply:
I think you are right. Given that I'd like to get this merged, I'll remove the `isPlainObject` guard for now. The `isObject` guard looks fine the way it is, and we can always make another PR that adds an `isPlainObject` guard, maybe with a similar implementation like [lodash](https://github.com/lodash/lodash/blob/2f79053d7bc7c9c9561a30dda202b3dcd2b72b90/isPlainObject.js) is it (but with better types)

Reply:
f793f762